binary_search_tree/tree.py

```diff
@@ -14,42 +17,117 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log(n))
+    # Space Complexity: O(1) 
```

Comment:
🪐 Space complexity will actually be O(log(n)) here because of the recursive call stack